### ISSUE TITLE
Improve accessibility of user status indicator for colorblind users

### ISSUE THREAD

**Issue by joshstrom** [RocketChat/Rocket.Chat#3625]
Your Rocket.Chat version: Windows App: 1.3.1, Server: Still an issue for me on the public demo server

As a colorblind user (red/green), I find it very difficult to determine the status of a user (available, away, etc.) from the small colored dot next to the name. Would it be possible to either customize per user which colors should be used, or, perhaps even easier, display a tooltip with the user's status when the mouse rolls over the status dot?

Thanks,
Josh

**Comment by sunnyolives** [comment]
Another suggestion is to increase redudancy of the existing dots, by using shapes or patterns. For instance, the [power](https://en.wikipedia.org/wiki/Power_symbol) symbol, where on is a line (1) and off is a circle (0).

**Comment by geekgonecrazy** [comment]
A tooltip here would probably be the easiest.  

@joshstrom do you think that is sufficient?  I don't know much about what colors are easier to distinguish?  If we need to go further maybe we can add some accessibility features for the user where we could switch things like this.   Do you happen to know of any standard alternative colors here?

**Comment by joshstrom** [comment]
@geekgonecrazy A tooltip would go a _really_ long way to improving the situation. As @sunnyolives mentioned, however, it is often better to rely not only on color to communicate this information, but also on the shape of the icons or other clues. 

![image](https://cloud.githubusercontent.com/assets/1166599/16426566/7b876dd8-3d1e-11e6-9eb1-e70b871bd0db.png)

Here is a somewhat decent example (stolen from [a blog post by Jen Underwood](http://www.jenunderwood.com/2014/09/04/visualcue/) of status indicators that use both shape and color differences to relay information. A final target of something like this would be ideal. 

For the short term (at least in my case), a tooltip would solve the problem. Personally, I only have issues distinguishing between the "Online" and "Away" states ("Busy" and "Invisible" are distinct enough that I can tell at a glance, although I cannot speak for everyone). 

Thanks!
Josh

**Comment by geekgonecrazy** [comment]
@joshstrom thanks for the additional information!

**Comment by spiltfish** [comment]
We've found that changing the online color to #4dff4d has made a big improvement for our colorblind users, if anyone is looking for a quick fix for their systems.

**Comment by geekgonecrazy** [comment]
@spiltfish Thanks for sharing that color value!

If we added an accessibility setting to users preferences for color blind friendly colors and when toggled changed the online color to: `#4dff4d` would this work?

**Comment by spiltfish** [comment]
@geekgonecrazy I'm not sure what type of color-blind our users are, but I think that would be a great starting point! Admittedly as a non-color-blind user, #4dff4d is a little abrasive.

**Comment by joshstrom** [comment]
As a color-blind person, I believe this would be enough of a change to make it easier for me, personally, to distinguish the statuses.

**Comment by geekgonecrazy** [comment]
Good deal.  I agree its abrasive for us.  But behind a preference for that user would be safe. 

I've added the `contrib: easy` label to this.  Maybe someone looking to start contributing can take a look.  Should be pretty easy.  Otherwise one of us will try and take a look when we get some free development cycles. 👍 

Thanks for the feedback guys!

**Comment by sandrabordini2** [comment]
![direct-1](https://cloud.githubusercontent.com/assets/20732772/23717107/531abdea-0411-11e7-9698-a8b814c47c9c.png)
![direct-2](https://cloud.githubusercontent.com/assets/20732772/23717109/53524b48-0411-11e7-853e-33e3c9bcc023.png)
![general-1](https://cloud.githubusercontent.com/assets/20732772/23717108/5342ec5c-0411-11e7-82cf-461f45fe7233.png)
![general-2](https://cloud.githubusercontent.com/assets/20732772/23717111/53705eda-0411-11e7-86e1-e04c8c25888e.png)

I created two options of a solution.
In the first option I used labels indicating if the user is online or offline which is better than tooltips. In the second option I used really discreet icons indicating if the user is online or offline. In both options I kept the used colors.

**Comment by joshstrom** [comment]
@sandrabordini2 - I love these!

Personally, I prefer your first example:
- While the "dots" in the second example _are_ different, it is not immediately obvious what the symbol in the middle means (does the "v" mean visible, or is it just the down-arrow symbol). You would probably have to have other documentation as to what the symbol means.
- The words would probably be easier to localize to other languages and cultures than a dot with a symbol/letter in the middle.

Also (and this is probably just personal to me), but when the color is on the '@' symbol, I can actually easily distinguish the color. Not sure why it is easier (or maybe I am just viewing on a different monitor). but it is interesting.

**Comment by karlprieb** [comment]
Hi guys, we are working on those different color and forms to improve accessibility on our [UX Review](https://github.com/RocketChat/Rocket.Chat/projects/4) :) 

We are still discussing about that, but we already has a preview.
Take a look
![image](https://cloud.githubusercontent.com/assets/9200155/23723733/13ce3528-0429-11e7-843d-78065422eecd.png)

What you guys think?

**Comment by marceloschmidt** [comment]
@sandrabordini2 do you mind creating a PR with your proposed changes? Thanks!!

**Comment by shivamluthra** [comment]
Hi! Any PR pending for review for this issue? Or should I have a look at this as I'm searching for an easier issue to contribute? A little guidance from where to get started would be awesome!

**Comment by jkshapiro** [comment]
Icons are an improvement over color alone (see https://www.w3.org/TR/WCAG21/#use-of-color for WCAG's guidance). We'll still need some text--hover or otherwise, as long as it's accessible--providing the meaning of each icon.

**Comment by vynmera** [comment]
Anyone working on this? Otherwise I can pick it up - it seems relatively simple to implement.

**Comment by jkshapiro** [comment]
Nice to have the icons on the dots, that helps. We could make them more visible by increasing the colour contrast; check out https://www.w3.org/TR/WCAG22/#non-text-contrast

**Comment by ivans-netto** [comment]
Closing since presence indicators were reimplemented following WCAG standards for color contrast.
